other/apidsl/toxav.in.h

```diff
@@ -0,0 +1,621 @@
+   * rates are 8000, 12000, 16000, 24000, or 48000.
+   */
+  bool send_frame(uint32_t friend_number, const int16_t *pcm, size_t sample_count, 
+                  uint8_t channels, uint32_t sampling_rate) with error for send_frame;
+}
+namespace video {
+  /**
+   * Send a video frame to a friend.
+   *
+   * Y - plane should be of size: height * width
+   * U - plane should be of size: (height/2) * (width/2)
+   * V - plane should be of size: (height/2) * (width/2)
+   *
+   * @param friend_number The friend number of the friend to which to send a video
+   * frame.
+   * @param width Width of the frame in pixels.
+   * @param height Height of the frame in pixels.
+   * @param y Y (Luminance) plane data.
+   * @param u U (Chroma) plane data.
+   * @param v V (Chroma) plane data.
+   * @param a A (Alpha) plane data.
+   */
+  bool send_frame(uint32_t friend_number, uint16_t width, uint16_t height,
+                  const uint8_t *y, const uint8_t *u, const uint8_t *v, const uint8_t *a) with error for send_frame;
+}
+/*******************************************************************************
+ * 
+ * :: A/V receiving
+ *
+ ******************************************************************************/
+namespace audio {
+  event receive_frame {
+    /**
+     * The function type for the ${event receive_frame} callback.
+     *
+     * @param friend_number The friend number of the friend who sent an audio frame.
+     * @param pcm An array of audio samples (sample_count * channels elements).
+     * @param sample_count The number of audio samples per channel in the PCM array.
+     * @param channels Number of audio channels.
+     * @param sampling_rate Sampling rate used in this frame.
+     *
+     */
+    typedef void(uint32_t friend_number, const int16_t *pcm, size_t sample_count,
+                 uint8_t channels, uint32_t sampling_rate);
+  }
+}
+namespace video {
+  event receive_frame {
+    /**
+     * The function type for the ${event receive_frame} callback.
+     *
+     * @param friend_number The friend number of the friend who sent a video frame.
+     * @param width Width of the frame in pixels.
+     * @param height Height of the frame in pixels.
+     * @param y 
+     * @param u 
+     * @param v Plane data.
+     *          The size of plane data is derived from width and height where
+     *          Y = MAX(width, ystride) * height, 
+     *          U = MAX(width/2, ustride) * (height/2) and 
+     *          V = MAX(width/2, vstride) * (height/2).
+     *          A = MAX(width, astride) * height.
+     * @param ystride
+     * @param ustride
+     * @param vstride 
+     * @param astride Strides data.
+     */
+    typedef void(uint32_t friend_number, uint16_t width, uint16_t height,
+                 const uint8_t *y, const uint8_t *u, const uint8_t *v, const uint8_t *a, 
+                 int32_t ystride, int32_t ustride, int32_t vstride, int32_t astride);
```

Comment:
It comes from the decoder. Whenever strides data is present it's absolute is always greater than width (or width/2).